airqo-platform/AirQo-api
# calibrate measurement

# Feature

**_WHAT DOES THIS PR DO?_**
Creates an endpoint for computing calibrated values using two calibration models.

**_WHAT JIRA STORY/TASK IS RELATED TO THIS PR?_**
https://airqoteam.atlassian.net/jira/software/projects/PLAT/boards/41?assignee=5d91eed256bb950dd6489270&selectedIssue=PLAT-394

**_WHAT IS THE LINK TO THE PR BRANCH_**
https://github.com/airqo-platform/AirQo-api/tree/ft-calibrate-measurement

**_HOW DO I TEST OUT THIS PR?_**
`cd AirQo-api/src/calibrate`
`python3 -m venv [DESTINATION_FOLDER]`
`source [DESTINATION_FOLDER]/bin/activate`
`pip install -r requirements/dev.txt`
`flask run`
Make sure you add a GCP service account key under jobs/

**_WHICH ENDPOINTS SHOULD BE READY FOR TESTING?:_** 
[POST] http://localhost:4001/api/v1/calibrate

Sample Input body - 
{
"datetime":"2020-09-06 12:56:59+00:00",
"raw_values": [
    {
    "raw_value": 100,
    "sensor_id": 832251
    },
    {
    "raw_value": 22,
    "sensor_id": 82720
    },
    {
    "raw_value": 29,
    "sensor_id": 870144
    }
            ]
}


**_IS THERE ANY JENKINS CONSOLE LINK FOR CODE COVERAGE AND BUILD INFO?_**
N/A

**_ARE THERE ANY RELATED PRs?_**
N/A
